doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -0,0 +1,947 @@
+* Both switch and the DPUs will follow to the [SONiC system health monitor HLD](https://github.com/sonic-net/SONiC/blob/ce313db92a694e007a6c5332ce3267ac158290f6/doc/system_health_monitoring/system-health-HLD.md)
+* Refer to section 3.4.5 for "show system-health .." CLIs
+
+##### 3.1.5.2 ModuleBase class new APIs
+The DPU ID is used only for indexing purpose.
+
+get_dpu_id(self):
+```
+    Retrieves the DPU ID. Returns None for non-smartswitch chassis.
+
+    Returns:
+        An integer, indicating the DPU ID. DPU0 returns 1, DPUX returns X-1
+```
+#### DPU_STATE Use Case
+* High Availability, Load Balancing, Debug, error recovery (reset, power cycle) and fault management logics will use this API
+
+get_state_info(self, index):
+```
+    Retrieves the dpu state object having the detailed dpu state progression.
+    Fetched from ChassisStateDB.
+
+    Returns:
+        An object instance of the DPU_STATE (see DB schema)
+        Returns None when the index is 0 (switch)
+```
+
+#### DPU_STATE Use Case
+* The major consumer of this data could be fault management, debug, error recovery (TBD)
+
+get_health_info(self, index):
+```
+    Retrieves the dpu health object having the detailed dpu health
+    Fetched from the DPUs
+
+    Returns:
+        An object instance of the dpu health
+        Returns None when the index is 0 (switch)
+```
+
+### 3.2 Thermal management
+* Platform  initializes all sensors
+* Thermalctld fetch CPU temperature, DPU temperature, fan speed, monitor and update the DB
+* Thermal manager reads all thermal sensor data, run thermal policy and take policy action Ex. Set fan speed, set alarm, set syslog, set LEDs 
+* Platform collects fan related data such as presence, failure and then applies fan algorithm to set the new fan speed
+* The north bound CLI/Utils/App use DB data to ”show environment”, ”show platform temp” show platform fan”
+
+Thermal management sequence diagram
+<p align="center"><img src="./images/thermal-mgmt-seq.svg"></p>
+
+#### 3.2.1 Platform device data collection 
+* thermalctld, led and PSUd post device data to DB periodically
+* during the boot up of the daemons, it will collect the constant data like serial number, manufacture name, etc.
+* For the variable ones (temperature, voltage, fan speed ....) need to be collected periodically. 
+
+### 3.3   Midplane Interface
+A typical modular chassis includes a midplane-interface to interconnect the Supervisor & line-cards. When DPU card or the Switch boots and as part of its initialization, midplane interface gets initialized.
+* Please refer to this [link](https://github.com/sonic-net/SONiC/blob/master/doc/smart-switch/ip-address-assigment/smart-switch-ip-address-assignment.md) for IP address assignment between the switch host and the DPUs.
+
+### 3.4 Debug & RMA
+CLI Extensions and Additions
+
+#### Platform CLIs
+* Platform CLIs remain the same but include DPU, and changes in sensors and FRUs
+
+show platform inventory 
```

Comment:
Is it a new command planned to be added in this HLD design? I didn't see it in the current SONiC code or I missed?

Reply:
The common API is "show platform ...".  The extension is platform specific.  This is an example to enhance the corresponding platform API to display the DPU modules. Updated the doc to show that it is an example.